client/tests/integration/smartcontracts/validator_with_custom_token/src/lib.rs

```diff
@@ -113,7 +116,10 @@ impl CustomValidator {
             .iter()
             .try_for_each(|(account, domain_id)| {
                 RevokeBox::new(
-                    PermissionToken::new(can_unregister_domain_definition_id.clone(), domain_id),
+                    PermissionToken::new(
+                        can_unregister_domain_definition_id.clone(),
+                        json!(domain_id),
```

Comment:
no big deal. Especially if you apply [this](https://github.com/hyperledger/iroha/pull/3775#discussion_r1281449756) comment. Then it's hidden away

Reply:
> it feels like we're tied to serde_json

Yes, as well, as we tied to `parity_scale_codec`. But I don't see much problems here.
The problem of keeping `String` is that it will be Serialized/Deserialized as string, not as JSON. See my response for Roman's comment

Reply:
let's move discussion to that comment